src/en/mangahen/src/eu/kanade/tachiyomi/extension/en/mangahen/MangaHen.kt

```diff
@@ -0,0 +1,172 @@
+package eu.kanade.tachiyomi.extension.en.mangahen
+
+import eu.kanade.tachiyomi.extension.all.mangahen.SortFilter
+import eu.kanade.tachiyomi.extension.all.mangahen.TextFilter
+import eu.kanade.tachiyomi.extension.all.mangahen.TypeFilter
+import eu.kanade.tachiyomi.extension.all.mangahen.getFilters
+import eu.kanade.tachiyomi.network.GET
+import eu.kanade.tachiyomi.source.model.FilterList
+import eu.kanade.tachiyomi.source.model.MangasPage
+import eu.kanade.tachiyomi.source.model.Page
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.source.online.HttpSource
+import eu.kanade.tachiyomi.util.asJsoup
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.Request
+import okhttp3.Response
+import org.jsoup.nodes.Element
+
+class MangaHen : HttpSource() {
+
+    override val name = "MangaHen"
+
+    override val baseUrl = "https://manga-hen.com"
+
+    private val advSearchURL = "$baseUrl/advanced-search"
+
+    override val lang = "en"
+
+    override val supportsLatest = true
+
+    override val client = network.cloudflareClient
+
+    // Popular
+    override fun popularMangaRequest(page: Int): Request {
+        return GET("$advSearchURL/?search=1&type=0&sort=1&page=$page", headers)
+    }
+
+    override fun popularMangaParse(response: Response): MangasPage {
+        val doc = response.asJsoup()
+
+        val mangas = doc.select("a[href^=/manga/]").map(::popularMangaFromElement)
+
+        val hasNextPage = doc.select("a[href*=page]").any { it.text().isEmpty() || it.text().isBlank() }
+
+        return MangasPage(mangas, hasNextPage)
+    }
+
+    private fun popularMangaFromElement(element: Element): SManga {
+        return SManga.create().apply {
+            title = element.selectFirst("h2")!!.ownText()
+            setUrlWithoutDomain(element.absUrl("href"))
+            thumbnail_url = element.selectFirst("img")!!.absUrl("src")
+        }
+    }
+
+    // Latest
+    override fun latestUpdatesRequest(page: Int): Request {
+        return GET("$advSearchURL/?search=1&type=0&sort=2&page=$page", headers)
+    }
+
+    override fun latestUpdatesParse(response: Response): MangasPage = popularMangaParse(response)
+
+    // Search
+
+    private fun tagSearch(tag: String): String? {
+        val index = (tagsList().indexOf(tag) + 1).toString()
+        return if (index != "-1") index else null
+    }
+
+    private fun tagsList(): List<String> {
+        val request = Request.Builder()
+            .url(advSearchURL)
+            .headers(headers)
+            .build()
+
+        val response = client.newCall(request).execute()
+        val tagsList = response.asJsoup().select("li[onclick=updateTag(this)]").map { it.ownText().lowercase() }
+
+        return tagsList
+    }

```

Comment:
fetch tag list once and cache them in a variable. currently you seem to be fetching them everytime in the loop for parsing search filters

or you can populat the filter list by fetching the tags too